File: tmp-mei/fgen4255027576711118260/base/EmptyMap_2.java
package base;
public interface EmptyMap_2 extends base.Map_2{
EmptyMap_2 $self = new EmptyMap_2Impl();
base.Opt_1 get$read(Object k_m$);

base.Opt_1 get$imm(Object k_m$);

base.Bool_0 keyEq$read(Object k1_m$, Object k2_m$);

base.Bool_0 isEmpty$read();
static base.Bool_0 keyEq$read$fun(Object k1_m$, Object k2_m$, base.EmptyMap_2 $this) {
  return ((base.Bool_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.Bool_0) null;
}})
);
}

static base.Opt_1 get$imm$fun(Object k_m$, base.EmptyMap_2 $this) {
  return ((base.Opt_1)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.Opt_1) null;
}})
);
}

static base.Opt_1 get$read$fun(Object k_m$, base.EmptyMap_2 $this) {
  return ((base.Opt_1)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.Opt_1) null;
}})
);
}
}

File: tmp-mei/fgenjs5030319945734616766/base/EmptyMap_2.js
import { base$$Map_2 } from "../base/Map_2.js";
import * as rt from "../rt/main.js";

export class base$$EmptyMap_2 extends base$$Map_2 {
  static $self = new base$$EmptyMap_2();

  get$read(k_m$) {
  return base$$EmptyMap_2.get$read$fun(k_m$, this);
}


get$imm(k_m$) {
  return base$$EmptyMap_2.get$imm$fun(k_m$, this);
}


keyEq$read(k1_m$, k2_m$) {
  return base$$EmptyMap_2.keyEq$read$fun(k1_m$, k2_m$, this);
}


isEmpty$read() {
  return base$$Map_2.isEmpty$read$fun(this);
}

  static keyEq$read$fun(k1_m$, k2_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static get$imm$fun(k_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static get$read$fun(k_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}
